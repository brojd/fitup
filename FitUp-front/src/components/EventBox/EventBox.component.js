import React, { PropTypes } from 'react';
import styles from './EventBox.stylesheet.css';
import Image from '../Image/Image.component';
import classNames from 'classnames';

class EventBox extends React.Component {
  
  constructor() {
    super();
    this._startTime = this._startTime.bind(this);
    this._endTime = this._endTime.bind(this);
  }
  
  _startTime() {
    let hours = (new Date(this.props.currentEvent.start_date)).getHours();
    let minutes = (new Date(this.props.currentEvent.start_date)).getMinutes();
    if (minutes.toString().length === 1) {
      minutes = '0'.concat(minutes);
    }
    return `${hours}:${minutes}`;
  }
  
  _endTime() {
    let hours = (new Date(this.props.currentEvent.end_date)).getHours();
    let minutes = (new Date(this.props.currentEvent.end_date)).getMinutes();
    if (minutes.toString().length === 1) {
      minutes = '0'.concat(minutes);
    }
    return `${hours}:${minutes}`;
  }
  
  render() {
    return (
        <aside style={this.props.eventVisible ? {top: this.props.eventTop, left: this.props.eventLeft} :
                                                {top: this.props.eventTop, left: 5000}}
               className={classNames(styles.eventDetails, 'uk-width-2-10',
               this.props.eventVisible ? 'fadeInFast' : 'hide')}>
          <div className={styles.classesImgWrapper}>
            <Image className={classNames('uk-width-9-10 uk-align-center', styles.classesImgWrapper__classesImg)}
                   src={this.props.currentClasses.photo_url}
                   alt={this.props.currentClasses.name}/>
            <div className={styles.classesImgWrapper__durationBack}>
              <span className={styles.classesImgWrapper__durationText}>
                {this.props.currentEvent.duration_in_minutes} min
              </span>
            </div>
            <div className={styles.classesImgWrapper__time}>
              {this._startTime()} - {this._endTime()}
            </div>
          </div>
          <div className={classNames(styles.classesInfoWrapper, 'uk-width-9-10 uk-align-center')}>
            <h3 className={classNames(styles.classesInfoWrapper__header,
                                      'uk-text-center uk-width-5-10 uk-align-center')}>
              {this.props.currentClasses.name}
            </h3>
            {this.props.children}
          </div>
        </aside>
    );
  }
}

EventBox.propTypes = {
  currentEvent: PropTypes.object,
  currentTrainer: PropTypes.object,
  currentClasses: PropTypes.object,
  currentGym: PropTypes.object,
  eventTop: PropTypes.number,
  eventLeft: PropTypes.number,
  eventVisible: PropTypes.bool,
  children: PropTypes.element
};

EventBox.defaultProps = {
  currentEvent: {},
  currentTrainer: {},
  currentClasses: {},
  currentGym: {
    address:{
      city:''
    },
    cards_accepted: [''],
    pricing: [{}]
  },
  eventTop: 100,
  eventLeft: 100,
  eventVisible: false
};

export default EventBox;
