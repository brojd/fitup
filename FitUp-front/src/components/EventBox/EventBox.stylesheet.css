.eventDetails {
  font-family: Century Gothic, CenturyGothic, AppleGothic, sans-serif !important;
  position: fixed;
  background: rgba(82, 82, 82, 0.86);
  border-radius: 1%;
  border: 1px solid #FFFFFF;
  color: #FFFFFF;
  z-index: 8; }

.classesImgWrapper {
  position: relative; }

.classesImgWrapper__classesImg {
  margin-top: 15px;
  border-radius: 1%;
  border: 1px solid rgba(255, 255, 255, 0.89); }

.classesImgWrapper__durationBack {
  position: absolute;
  bottom: 10px;
  left: 25px;
  background: rgba(189, 0, 0, 0.92);
  width: 60px;
  height: 60px;
  border-radius: 50%; }

.classesImgWrapper__durationText {
  position: absolute;
  bottom: 21px;
  left: 5px; }

.classesImgWrapper__time {
  position: absolute;
  right: 30px;
  height: 20px;
  padding: 10px;
  bottom: 15px;
  background: rgba(189, 0, 0, 0.92); }

.classesInfoWrapper {
  color: #FFFFFF; }

.classesInfoWrapper__header {
  background: rgba(189, 0, 0, 0.96);
  text-transform: uppercase;
  font-size: 18px;
  font-style: italic;
  font-weight: 600;
  color: #FFFFFF; }

/*# sourceMappingURL=EventBox.stylesheet.css.map */
