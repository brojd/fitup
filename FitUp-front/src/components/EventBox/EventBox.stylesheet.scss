@import "../../../sass_variables/colors";

.eventDetails {
  font-family: Century Gothic, CenturyGothic, AppleGothic, sans-serif !important;
  position: fixed;
  background: rgba(82, 82, 82, 0.86);
  border-radius: 1%;
  border: 1px solid $white;
  color: $white;
  z-index: 8;
}

.classesImgWrapper {
  position: relative;
}

.classesImgWrapper__classesImg {
  margin-top: 15px;
  border-radius: 1%;
  border: 1px solid rgba($white, 0.89);
}

.classesImgWrapper__durationBack {
  position: absolute;
  bottom: 10px;
  left: 25px;
  background: rgba($guardsman_red, 0.92);
  width: 60px;
  height: 60px;
  border-radius: 50%;
}

.classesImgWrapper__durationText {
  position: absolute;
  bottom: 21px;
  left: 5px;
}

.classesImgWrapper__time {
  position: absolute;
  right: 30px;
  height: 20px;
  padding: 10px;
  bottom: 15px;
  background: rgba($guardsman_red, 0.92);
}

.classesInfoWrapper {
  color: $white;
}

.classesInfoWrapper__header {
  background: rgba($guardsman_red, 0.96);
  text-transform: uppercase;
  font-size: 18px;
  font-style: italic;
  font-weight: 600;
  color: $white;
}
