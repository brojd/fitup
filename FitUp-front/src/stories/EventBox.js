import React from 'react';
import { storiesOf} from '@kadira/storybook';
import EventBox from '../components/EventBox/EventBox.component';
import EventBoxList from '../components/EventBoxList/EventBoxList.component';
import event from './mock_data/eventInstance';

storiesOf('EventBox', module)
  .add('event box', () => (
    <EventBox { ...event }>
      <EventBoxList currentTrainer={event.currentTrainer} currentGym={event.currentGym} />
    </EventBox>
  ));

