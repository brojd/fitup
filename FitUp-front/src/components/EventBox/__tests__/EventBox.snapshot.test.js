import React from 'react';
import EventBox from '../EventBox.component';
import renderer from 'react-test-renderer';

test('EventBox renders correctly', () => {
  
  const EVENT_TOP = 580;
  const EVENT_LEFT = -200;
  const EVENT_VISIBLE = true;
  const CURRENT_EVENT = {
    '_id': 1479128395909,
    'start_date': '2016-11-14T14:50:00.000Z',
    'end_date': '2016-11-14T15:40:00.000Z',
    'duration_in_minutes': 50,
    'classesName': 'yoga',
    'trainerName': 'Andrew Kowalski',
    'gymId': {
      '$oid': '581c7f51d686e4e41104eeb1'
    },
    'typeOfClassesId': {
      '$oid': '581cac9645633510184e0b80'
    },
    'trainerId': {
      '$oid': '581cacc145633510184e0b84'
    },
    'text': '',
    '_timed': true,
    '_sday': 0,
    '_inner': false,
    '_sorder': 0,
    '_count': 1,
    'color': '#661141'
  };
  const CURRENT_TRAINER = {
    '_id': {
      '$oid': '581cacc145633510184e0b84'
    },
    'name': 'Andrew',
    'surname': 'Kowalski',
    'about_me': 'Lorem ipsum',
    'specialties': {
      'yoga': true,
      'zumba': false
    },
    'photo_url': 'http://trenujzmichalem.pl/wp-content/uploads/2014/01/ostr-slider-535x572.png',
    'gymId': '581c7f51d686e4e41104eeb1',
    'id': {
      '$oid': '581cacc145633510184e0b84'
    }
  };
  const CURRENT_CLASSES = {
    '_id': {
      '$oid': '581cada645633510184e0b88'
    },
    'name': 'pilates',
    'description': 'pilates description',
    'photo_url': 'http://fitnessforladies.pl/wp-content/uploads/2013/12/pilates-640x504.jpg'
  };
  const CURRENT_GYM = {
    '_id': {
      '$oid': '581ca32345633510184e0b5d'
    },
    'name': 'Fitness World',
    'address': {
      'city': 'Wroc≈Çaw',
      'postal_code': '50-001',
      'street': 'Olszowa',
      'number': 3
    },
    'description': 'Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
    'logo_url': 'http://www.feriogaj.com/wp-content/uploads/fitnes-world.png',
    'photo_urls': [
      'http://www.fru-fitness.pl/content/57/0/1370494719_6ca505f63f8064d96281ac502dd903f2.jpg',
      'http://www.top-gym.pl/files/dzialy/wydarzenia/lipiec%202015/Art%20Fitness2.png'
    ],
    'pricing': [
      {
        'name': '1 workout',
        'price': 15
      },
      {
        'name': '1 hour',
        'price': 40
      },
      {
        'name': '1 month',
        'price': 130
      }
    ],
    'cards_accepted': [
      {
        'name': 'Multisport',
        'accepted': false
      },
      {
        'name': 'ActiveSport',
        'accepted': false
      },
      {
        'name': 'OKsystem',
        'accepted': false
      },
      {
        'name': 'FitWro',
        'accepted': true
      },
      {
        'name': 'FitProfit',
        'accepted': false
      }
    ],
    'password': '$2a$10$ZqmPnzKm4QU5WRrZTkrhbup8UpDM55p2wreARh6cCc6FpUF3vv9Xa',
    'email': 'gym2@gym2.pl',
    'id': {
      '$oid': '581ca32345633510184e0b5d'
    }
  };
  
  const component = renderer.create(
    <EventBox eventTop={EVENT_TOP}
              eventLeft={EVENT_LEFT}
              eventVisible={EVENT_VISIBLE}
              currentGym={CURRENT_GYM}
              currentTrainer={CURRENT_TRAINER}
              currentClasses={CURRENT_CLASSES}
              currentEvent={CURRENT_EVENT} />
  );
  
  let tree = component.toJSON();
  
  expect(tree).toMatchSnapshot();
  
});
