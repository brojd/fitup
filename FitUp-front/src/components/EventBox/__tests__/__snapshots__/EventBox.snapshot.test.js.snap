exports[`test EventBox renders correctly 1`] = `
<aside
  className="eventDetails uk-width-2-10 fadeInFast"
  style={
    Object {
      "left": -200,
      "top": 580,
    }
  }>
  <div
    className="classesImgWrapper">
    <img
      alt="pilates"
      className="uk-width-9-10 uk-align-center classesImgWrapper__classesImg"
      height={150}
      onClick={undefined}
      src="http://fitnessforladies.pl/wp-content/uploads/2013/12/pilates-640x504.jpg"
      width={150} />
    <div
      className="classesImgWrapper__durationBack">
      <span
        className="classesImgWrapper__durationText">
        50
         min
      </span>
    </div>
    <div
      className="classesImgWrapper__time">
      15:50
       - 
      16:40
    </div>
  </div>
  <div
    className="classesInfoWrapper uk-width-9-10 uk-align-center">
    <h3
      className="classesInfoWrapper__header uk-text-center uk-width-5-10 uk-align-center">
      pilates
    </h3>
  </div>
</aside>
`;
